src/ui/components/results/verticalresultscountcomponent.js

```diff
@@ -1,10 +1,15 @@
 import Component from '../component';
 import StorageKeys from '../../../core/storage/storagekeys';
+import SearchStates from '../../../core/storage/searchstates';
 
 export default class VerticalResultsCountComponent extends Component {
   constructor (config = {}, systemConfig = {}) {
     super(config, systemConfig);
-    this.moduleId = StorageKeys.RESULTS_HEADER;
+    this.core.globalStorage.on('update', StorageKeys.VERTICAL_RESULTS, results => {
+      if (results.searchState === SearchStates.SEARCH_COMPLETE) {
+        this.setState();
```

Comment:
Is this setState expecting data?

Reply:
no, no data

Reply:
This may be never a case currently, but it seems weird that we are listening for a `VERTICAL_RESULTS` update but not using it and instead re-fetching it in `setState`. It makes me feel like there is a case when it might change in between. Though I'm not sure that would be a problem? What do you think?

Reply:
javascript is single threaded so that wouldn't ever be an issue. this is also how the pagination component works

Reply:
I'm not sure -- what about

AnotherFile.js
```
this.core.globalStorage.on('update', StorageKeys.VERTICAL_RESULTS, results => {
  this.core.globalStorage.setState(StorageKeys.VERTICAL_RESULTS, 'update it for some reason')
});
```

verticalresultscountcomponent.js
```
this.core.globalStorage.on('update', StorageKeys.VERTICAL_RESULTS, results => {
  this.setState() // getState !== results
});
```

Reply:
I'm not saying that's a case that should ever happen* Just feels like something we can avoid ourselves from if we passed exactly what we're getting from update?

Reply:
Sure, I can update it if that puts you at ease. I don't see how example you gave is an issue, though, partially because it's an infinite loop but mostly because even if it was just a regular globalStorage setState I don't think there would be any issues?

Reply:
if AnotherComponent updated vertical results, wouldn't the order of execution just be

AnotherComponent updates vertical results -> this triggers all the event listeners, which run synchronously in the order they were registered -> whatever code is after that in AnotherComponent

Reply:
this makes things easier to unit test though which is kind of nice

Reply:
womp-- Sorry, infinite loop, you're right

Reply:
this makes things a little nicer to unit test, but it's a little weird that it relies on two pieces of information (search offset and vertical results) but only one gets to be the special "data" param

Reply:
I think you're right though this is more idiomatic of the SDK (even though I don't really like it)